Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -178,18 +178,30 @@ private static IEnumerable<BusinessObjectValidationFailure> CreateBusinessObject
       return result;
     }
 
-    public IReadOnlyCollection<UnhandledBusinessObjectValidationFailure> GetUnhandledValidationFailures (
-        IBusinessObject businessObject,
-        bool includePartiallyHandledFailures = false)
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailures (
+        [NotNull] IBusinessObject businessObject,
+        bool includePartiallyHandledFailures = false,
+        bool markAsHandled = false)
     {
-      ArgumentUtility.CheckNotNull("businessObject", businessObject);
+      ArgumentUtility.CheckNotNull(nameof(businessObject), businessObject);
 
-      return _unhandledValidationFailures
+      var validationFailures = _validationFailures
```

Comment:
This one here changes the semantic. we need to keep using _unhandledValidationFailures to only get those that are still unhandled.

The intended semantic of the originel code was:
One ValidationFailure can result in multiple BusinessObjectValidationFailures.
include-partially-handled=true means we return BusinessObjectValidationFailures where any part of ValidationFailure was already handled.

With the semantic as it is written in the updated version, we use the complete set of known BusinessObjectValidationFailures as the jump-off point, including those already addressed by an earlier call.

Unless you already know the answer, we may need to discuss with @a-ctor the intended semantic to clear this up.